ansible-tests/validations/check_openstack_endpoints.yaml

```diff
@@ -0,0 +1,35 @@
+---
+- hosts: undercloud
+  vars:
+    metadata:
+      name: Check access to various openstack services
+      description: >
+        This will attempt to connect to each service endpoint defined in
+        your hosts file's controller_vip variable.
+      groups:
+      - post-deployment
+  tasks:
+  - name: Load the overcloud credentials
+    overcloudrc: path=/home/stack/overcloudrc
+
+  # Check connectivity to horizon
+  - name: Check Horizon
+    # TODO: get Horizon endpoint from Keystone
+    uri: url=http://{{public_vip}}
+
+  # Check that we can obtain an auth token from horizon
+  - name: Check Keystone
+    uri:
+     url: "{{ overcloudrc.OS_AUTH_URL }}/tokens"
+     method: POST
+     body_format: json
+     body:
+       auth:
+         passwordCredentials:
+           username: "{{ overcloudrc.OS_USERNAME }}"
```

Comment:
What happens if the OS_USERNAME variable is not defined in `~stack/overcloudrc`?